lib/spack/llnl/util/lang.py

```diff
@@ -442,6 +442,59 @@ def pretty_date(time, now=None):
         return str(diff) + " years ago"
 
 
+def str2date(date_str):
```

Comment:
How about `pretty_str_to_date` or something to indicate it's reciprocal?  Also needs a test case (can probably reuse the cases for `pretty_date`)